pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
@abronan not sure what you think, but this might be an area we could improve libkv's impl to try to union the TLS settings together instead of clobbering them, although that might get messy, so it seems like the right answer might be pushing settings all the way to the etcd go library.